### ISSUE TITLE
In-score text entry cursor scales incorrectly (and is generally poorly sized)

### ISSUE THREAD

**Issue by avvvvve** [musescore/MuseScore#26218]
### Issue type

UI bug (incorrect info or interface appearance)

### Description with steps to reproduce

When **Page Settings > Scaling > Staff space (sp)** is small (i.e. 0.040), the text entry cursor looks comically wide. Its width could instead scale with the 'Staff space' setting.

The height of the cursor does _not_ scale for dynamics text boxes, but it does (correctly) for other types of text.

_____

And to make it generally look prettier:

- The cursor should be slightly larger than the font's cap height and vertically centered to it (right now it's too short)
- The cursor could be slightly less wide by default
- [It should blink](https://github.com/musescore/MuseScore/issues/24808) (already logged)

### Supporting files, videos and screenshots

https://github.com/user-attachments/assets/32d2ba8f-cbef-452e-b48a-6fc7036daa07

### What is the latest version of MuseScore Studio where this issue is present?

Nightly

### Regression

No.

### Operating system

macOS 15

### Checklist

- [x] This report follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting bugs and issues
- [x] I have verified that this issue has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar issues
- [x] I have attached all requested files and information to this report
- [x] I have attempted to identify the root problem as concisely as possible, and have used minimal reproducible examples where possible

**Comment by krasko78** [comment]
Doesn't it make more sense for the width of the cursor to depend on the size of the text being edited: larger text -> wider cursor? (If "scale with staff size" is enabled for the text being edited, then the width of the cursor will indirectly depend on the staff space.)

**Comment by avvvvve** [comment]
Yup, that makes sense. I just edited the description accordingly.
